src/game/BattleGround/BattleGround.cpp

```diff
@@ -1729,8 +1729,12 @@ void BattleGround::HandleTriggerBuff(ObjectGuid go_guid)
 */
 void BattleGround::HandleKillPlayer(Player* player, Player* killer)
 {
-    // add +1 deaths
-    UpdatePlayerScore(player, SCORE_DEATHS, 1);
+    if (!player->HasAura(27827)) // do not count spirit of redemption
```

Comment:
I am a bit late on this one, but we should perhaps count the spirit of redemption one, and then dont count the instakill one. Cos player instakills himself at the end of spirit of redemption and i dont think that should be counted per BG screen.